# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp
# Commit: b0cfee4dccb7
# Full Hash: b0cfee4dccb70dbb0fd958d7430e1d0ce11fbdd8
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-18 21:32:34
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r 8973242af52a -r b0cfee4dccb7 dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp
--- a/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp	Tue Mar 18 12:10:53 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp	Tue Mar 18 12:10:54 2025 +0000
@@ -43,15 +43,15 @@
   RefPtr<WorkerDebuggerManager> manager = WorkerDebuggerManager::Get();
   MOZ_ASSERT_DEBUG_OR_FUZZING(manager);
 
-  nsCOMPtr<nsIWorkerDebugger> debugger =
-      manager->GetDebuggerById(aDebuggerInfo.Id());
-  MOZ_ASSERT_DEBUG_OR_FUZZING(!debugger);
-
-  debugger = MakeRefPtr<RemoteWorkerDebuggerParent>(aDebuggerInfo,
-                                                    std::move(aParentEp));
+  RefPtr<RemoteWorkerDebuggerParent> debugger =
+      MakeRefPtr<RemoteWorkerDebuggerParent>(aDebuggerInfo,
+                                             std::move(aParentEp));
 
   manager->RegisterDebugger(debugger);
 
+  MOZ_ASSERT(debugger->CanSend());
+  Unused << debugger->SendRegisterDone();
+
   return IPC_OK();
 }
 